### PR TITLE
oop assingments

### PR DESCRIPTION
**Pull request by codedeshi**

### CODE REVIEW

File: Hardit_Singh/bankAccount.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+				:userName,
+				:checkingBalance,
+				:savingBalance
+
+	def initialize(userName,balance=0)		
+		@accountNumber = genAccountNumber
+		@userName = userName
+		@checkingBalance = balance
+		@savingBalance = balance
+		@@noOfAccounts += 1
```
Comment (digitalecho) [inline Hardit_Singh/bankAccount.rb:12]: Does this error out because you don't initialize it as a number?

File: Hardit_Singh/bankAccount.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		self
+	end
+
+	def get(req)
+		eval"@#{req}"
+	end
+
+	def deposit(amount,acctype)
+		(['checking','saving'].include? acctype) ?  (eval"@#{acctype}Balance+=#{amount}") : 'Not a valid account type'
```
Comment (digitalecho) [inline Hardit_Singh/bankAccount.rb:21]: Ternaries are for simple operations or variable settings. That eval is jusy messy. A quick exit with a return checking for the include is better.

File: Hardit_Singh/bankAccount.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		self
+	end
+
+	def total
+		puts "Total Balance: #{@checkingBalance+@savingBalance}"
+	end
+
+	def withdraw(amount,acctype)
+		# i know this is evil but just wanted to see how nested ternary operators work!
+		(!['checking','saving'].include? acctype) ?  "wrong account type" : (eval"@#{acctype}Balance>=#{amount}") ? (eval"@#{acctype}Balance-=#{amount}") : "Insufficient Funds"
```
Comment (digitalecho) [inline Hardit_Singh/bankAccount.rb:31]: A nested ternary is the evil of all evils, now that you see that I expect you to never do it again.

File: Hardit_Singh/human.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+	def stats
+		puts "stealth #{@stealth}"
+		puts "strength #{@strength}"
+		puts "intelligence #{@intelligence}"
+		puts "health #{@health}"
+	end
+
+	def attack(obj)
+		obj.health-=3 if obj.class.ancestors.inspect.include? 'Human'
```
Comment (digitalecho) [inline Hardit_Singh/human.rb:23]: Don't need the inspect, String.class.ancestors.include?(Human) should work.

File: Hardit_Singh/mathDojo.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
+class MathDojo
+	attr_accessor :result
+	def initialize(result=0)
+		self.result = result
+	end
+
+	def mAdd(*args)
+		args = flat(args)
+		for i in 0...args.length
```
Comment (digitalecho) [inline Hardit_Singh/mathDojo.rb:9]: What if its not an array?

File: Hardit_Singh/bankAccount.rb (hunk 1 above)
Comment (codedeshi) [reply Hardit_Singh/bankAccount.rb:12]: Do you mean @@noOfAccounts? 
it doesnt give any errors. And I initiated it as @@noOfAccounts = 0.

File: Hardit_Singh/human.rb (hunk 4 above)
Comment (codedeshi) [reply Hardit_Singh/human.rb:23]: Lol forgot to remove inspect ..had it in while troubleshooting.

File: Hardit_Singh/mathDojo.rb (hunk 5 above)
Comment (codedeshi) [reply Hardit_Singh/mathDojo.rb:9]: since in only used *args I think everything goes in as an array

File: Hardit_Singh/bankAccount.rb (hunk 1 above)
Comment (digitalecho) [reply Hardit_Singh/bankAccount.rb:12]: Ah I didn't see it down there.

**Pull request merged**